test/extensions/stats_sinks/hystrix/hystrix_test.cc

```diff
@@ -0,0 +1,370 @@
+    Buffer::OwnedImpl buffer;
+    auto encode_callback = [&buffer](Buffer::Instance& data, bool) {
+      buffer.add(
+          data); // This will append to the end of the buffer, so multiple calls will all be dumped
+      // one after another into this buffer. See Buffer::Instance for other buffer
+      // buffer modification options.
+    };
+    ON_CALL(callbacks_, encodeData(_, _)).WillByDefault(Invoke(encode_callback));
+
+    return buffer;
+  }
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks_;
+  NiceMock<Server::MockInstance> server_;
+  NiceMock<Upstream::MockCluster> cluster1_;
+  Upstream::ClusterManager::ClusterInfoMap cluster_map_;
+
+  std::unique_ptr<HystrixSink> sink_;
+  NiceMock<Stats::MockSource> source_;
+  NiceMock<Upstream::MockClusterManager> cluster_manager_;
+
+  Upstream::MockClusterInfo* cluster1_info_ = new NiceMock<Upstream::MockClusterInfo>();
+  Upstream::ClusterInfoConstSharedPtr cluster1_info_ptr_{cluster1_info_};
+
+  NiceMock<Stats::MockStore> stats_store_1_;
+  NiceMock<Stats::MockStore> cluster1_stats_scope_;
+  const std::string cluster1_name_{"test_cluster1"};
+
+  NiceMock<Stats::MockGauge> membership_total_gauge_;
+  NiceMock<Stats::MockCounter> success_counter_;
+  NiceMock<Stats::MockCounter> error_5xx_counter_;
+  NiceMock<Stats::MockCounter> retry_5xx_counter_;
+  NiceMock<Stats::MockCounter> error_4xx_counter_;
+  NiceMock<Stats::MockCounter> retry_4xx_counter_;
+};
+
+TEST_F(HystrixSinkTest, EmptyFlush) {
+  InSequence s;
+  Buffer::OwnedImpl buffer = createClusterAndCallbacks();
+  // register callback to sink
+  sink_->registerConnection(&callbacks_);
+
+  sink_->flush(source_);
+
+  std::string data_message = TestUtility::bufferToString(buffer);
+  EXPECT_EQ(getStreamField(data_message, "errorPercentage"), "0");
+  EXPECT_EQ(getStreamField(data_message, "errorCount"), "0");
+  EXPECT_EQ(getStreamField(data_message, "requestCount"), "0");
+  EXPECT_EQ(getStreamField(data_message, "rollingCountSemaphoreRejected"), "0");
+  EXPECT_EQ(getStreamField(data_message, "rollingCountSuccess"), "0");
+  EXPECT_EQ(getStreamField(data_message, "rollingCountTimeout"), "0");
+}
+
+TEST_F(HystrixSinkTest, BasicFlow) {
+  InSequence s;
+  Buffer::OwnedImpl buffer = createClusterAndCallbacks();
+  // register callback to sink
+  sink_->registerConnection(&callbacks_);
+
+  for (int i = 0; i < 12; i++) {
+    buffer.drain(buffer.length());
+    ON_CALL(error_5xx_counter_, value()).WillByDefault(Return((i + 1) * 17));
+    ON_CALL(success_counter_, value()).WillByDefault(Return((i + 1) * 7));
+    cluster1_info_->stats().upstream_rq_timeout_.add(3);
+    cluster1_info_->stats().upstream_rq_pending_overflow_.add(8);
+    sink_->flush(source_);
+  }
+
+  //  //std::string rolling_map = sink_->getStats().printRollingWindow();
+  std::string rolling_map = sink_->printRollingWindows();
+  std::size_t pos = rolling_map.find("test_cluster1.total");
+  EXPECT_NE(std::string::npos, pos);
```

Comment:
nit: you could do this in one line:

EXPECT_NE(std::string::npos, rolling_map.find("test_cluster1.total"));

the advantage is that if it fails the test error message would be a little easier to debug because it would include the source of the find call.